test/Tester/CancellationTests/CancellationTokenPassingTests.cs

```diff
@@ -0,0 +1,161 @@
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Orleans;
+using Orleans.TestingHost;
+using Orleans.Threading;
+using UnitTests.GrainInterfaces;
+using Xunit;
+using UnitTests.Tester;
+
+namespace UnitTests.MembershipTests
+{
+    public class CancellationTokenPassingTests : HostedTestClusterPerTest
+    {
+        private readonly double[] cancellationDelaysInMS = { 0, 10, 50, 500 };
+
+        public override TestingSiloHost CreateSiloHost()
+        {
+            return new TestingSiloHost(new TestingSiloOptions
+            {
+                StartFreshOrleans = false,
+                StartPrimary = true,
+                StartSecondary = true,
+                AdjustConfig = config =>
+                {
+                    config.Globals.DefaultPlacementStrategy = "ActivationCountBasedPlacement";
+                }
+            });
+        }
+
+        [Fact, TestCategory("Functional")]
+        public async Task GrainTaskCancellation()
+        {
+            foreach (var delay in cancellationDelaysInMS.Select(TimeSpan.FromMilliseconds))
```

Comment:
use `Theory` instead of `Fact` to run multiple variations of the same tests. There should be a few examples in our codebase. We'll start using it more and more (since this is new to xUnit)